### PR TITLE
Issue-81: Enhancements on the onboarding flow 

### LINKED ISSUE #81: Enhancements on the onboarding flow
### Describe your suggested feature

The expected design for the onboarding flow:

1_figma
2_figma
3_figma
4_figma
success_figma

The onboarding flow design as implemented today:

1_today
2_real
3_real
4_real

The goal is to match the app with the [figma's specs ](https://www.figma.com/design/sUbcDnjC0W26wAzRpMNFTz/Invoice-Generator?node-id=1318-17543&t=hCBDWkvQq0Tv3C06-4)

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[version](https://github.com/theolm/ambush-app/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by alexrodrigues**
There were some minor adjustments to the onboarding flow and the success screen. This PR includes all these changes.

Android (Small Phone):

android_small_3
android_small_success
android_small_4
android_small_1
android_small_2

iPad Success:

Simulator Screenshot - iPad (10th generation) - 2024-11-14 at 12 25 34

iPhone Success: 

Simulator Screenshot - iPhone 16 - 2024-11-14 at 12 33 28

Happy Path:

https://github.com/user-attachments/assets/d8b410d7-28a7-455d-b2b5-f41dddf362e9

I will do this later:

https://github.com/user-attachments/assets/157500f1-b125-4a48-a69f-65f40a9d611c

Sad Path:

https://github.com/user-attachments/assets/77d7b2c1-38d1-423f-9b05-0f9c567fd26b

New Constrained Scaffold:

https://github.com/user-attachments/assets/023ed131-4f57-4c39-af77-c64b1d20cf49

https://github.com/user-attachments/assets/31ff5034-2123-413e-9cdf-fcb475a25dd1

closes issue #81

### CODE REVIEW

File: ios/Runner/AppDelegate.swift [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import UIKit
 import Flutter
 
-@UIApplicationMain
+@main
```
Comment (alexrodrigues) [inline ios/Runner/AppDelegate.swift:4]: After Swift 5.3 we should use this annotation, this change will ensure compatibility with Swift UI 😄

File: lib/src/core/routes/app_route.dart [hunk 2]
```diff
@@ -19,5 +19,6 @@ class AppRouter extends RootStackRouter {
     AutoRoute(page: AddInvoiceRoute.page),
     AutoRoute(page: OnBoardingRoute.page),
     AutoRoute(page: AddressRoute.page),
+    AutoRoute(page: OnboardingSuccessRoute.page),
```
Comment (pnalvarez) [inline lib/src/core/routes/app_route.dart:22]: 🔥

File: lib/src/presentation/settings/success/onboarding_success_screen.dart [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import 'package:ambush_app/src/core/routes/app_route.gr.dart';
+import 'package:auto_route/auto_route.dart';
+import 'package:ambush_app/src/designsystem/buttons.dart';
+import 'package:ambush_app/src/presentation/add_invoice/add_invoice_navigation_flow.dart';
+import 'package:flutter/material.dart';
+
+@RoutePage()
+class OnboardingSuccessScreen extends StatelessWidget {
```
Comment (pnalvarez) [inline lib/src/presentation/settings/success/onboarding_success_screen.dart]: Minor: I think the project is adopting `Page` for screen suffix as a pattern,

File: lib/src/presentation/settings/success/onboarding_success_screen.dart [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  Widget build(BuildContext context) {
+    final textTheme = Theme.of(context).textTheme;
+    final colorTheme = Theme.of(context).colorScheme;
+
+    return Scaffold(
+      appBar: AppBar(
+        leading: IconButton(
+          icon: Icon(Icons.arrow_back),
+          onPressed: () {
+            Navigator.pop(context);
```
Comment (pnalvarez) [inline lib/src/presentation/settings/success/onboarding_success_screen.dart]: Isn't the screen supposed to rely on its `flow navigator` to pop the context? I think we only call the `Navigator` for alert dialogs.

File: lib/src/presentation/settings/success/onboarding_success_screen.dart [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                    child: Text(
+                      "I'll do it later",
+                      style: TextStyle(
+                        color: colorTheme.primary,
+                        fontSize: 16,
+                      ),
+                    ),
+                  ),
+                  SizedBox(
+                    width: 16.0,
```
Comment (theolm) [inline lib/src/presentation/settings/success/onboarding_success_screen.dart]: Check the file lib/src/core/settings/const.dart

**Review by theolm — COMMENTED** [review]
Nice Pr.
I would suggest to also address the maxWidth of the main content.
I would create a custom scaffold for the screens to use, This way it will be easier to maintain.

File: lib/src/presentation/settings/success/onboarding_success_screen.dart (hunk 3 above)
Comment (alexrodrigues) [reply lib/src/presentation/settings/success/onboarding_success_screen.dart]: Fixed!!

File: lib/src/presentation/settings/success/onboarding_success_screen.dart (hunk 4 above)
Comment (alexrodrigues) [reply lib/src/presentation/settings/success/onboarding_success_screen.dart]: Fixed!!

File: lib/src/presentation/settings/success/onboarding_success_screen.dart (hunk 5 above)
Comment (alexrodrigues) [reply lib/src/presentation/settings/success/onboarding_success_screen.dart]: Hey Theo, I really appreciate your feedback! Could you please help me understand the "custom scaffold" part?
By the way, I added the margins from the const.dart file and also included the maxWidth constraints

Comment (theolm) [reply lib/src/presentation/settings/success/onboarding_success_screen.dart]: Your solution of maxWidth only works for the OnboardingSuccessPage. Every other page will not have a good experience in wide screens.

I'm suggesting you to create a custom component based on the scaffold (e.g. ConstrainedScaffold) that accepts a max width as a parameter. This way you can replace the current scaffold for the ConstrainedScaffold and fix all the screens at once.

Comment (alexrodrigues) [reply lib/src/presentation/settings/success/onboarding_success_screen.dart]: Got it! Thanks for the explanation, I will work on that!

Comment (alexrodrigues) [reply lib/src/presentation/settings/success/onboarding_success_screen.dart]: I’ve just created the ConstrainedScaffold and replaced it in each Scaffold across the Ambush App! Please let me know if this commit makes sense [commit](https://github.com/GetAmbush/ambush-app/pull/84/commits/b632169263193914a65a3101df2e2119d732992e)

File: lib/src/designsystem/constrained_scaffold.dart [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      floatingActionButton: widget.floatingActionButton,
+      body: Center(
+        child: ConstrainedBox(
+          constraints: BoxConstraints(maxWidth: widget.maxWidth),
+          child: widget.body,
+        ),
+      ),
+    );
+  }
+}
```
Comment (theolm) [inline lib/src/designsystem/constrained_scaffold.dart]: I'm afraid that's not right. This custom component must not remove the capabilities of its parent. The scaffold accepts multiple parameters to customize its behaviour. The custom scaffold must accept the same parameters.

scafold params
```dart
super.key,
    this.appBar,
    this.body,
    this.floatingActionButton,
    this.floatingActionButtonLocation,
    this.floatingActionButtonAnimator,
    this.persistentFooterButtons,
    this.persistentFooterAlignment = AlignmentDirectional.centerEnd,
    this.drawer,
    this.onDrawerChanged,
    this.endDrawer,
    this.onEndDrawerChanged,
    this.bottomNavigationBar,
    this.bottomSheet,
    this.backgroundColor,
    this.resizeToAvoidBottomInset,
    this.primary = true,
    this.drawerDragStartBehavior = DragStartBehavior.start,
    this.extendBody = false,
    this.extendBodyBehindAppBar = false,
    this.drawerScrimColor,
    this.drawerEdgeDragWidth,
    this.drawerEnableOpenDragGesture = true,
    this.endDrawerEnableOpenDragGesture = true,
    this.restorationId,
```

File: lib/src/presentation/settings/base_settings_page.dart [hunk 7]
```diff
@@ -21,8 +22,8 @@ class BaseSettingsPage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final textTheme = Theme.of(context).textTheme;
-    final colorScheme = Theme.of(context).colorScheme;
-    return Scaffold(
+    return ConstrainedScaffold(
+      maxWidth: double.infinity,
```
Comment (theolm) [inline lib/src/presentation/settings/base_settings_page.dart]: The `maxWidth` is probably not infinity. I think it should be defaultPageMaxWidth.

File: lib/src/designsystem/constrained_scaffold.dart [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  final AppBar? appBar;
+  final double maxWidth;
+  final Widget? body;
+  final Widget? floatingActionButton;
+
+  const ConstrainedScaffold({
+    super.key,
+    required this.appBar,
+    required this.maxWidth,
+    required this.body,
```
Comment (theolm) [inline lib/src/designsystem/constrained_scaffold.dart]: Follow the same pattern as the Scaffold. I think this should not be required.

btw. the default maxWidth should be infinity.

File: lib/src/presentation/settings/success/onboarding_success_screen.dart [hunk 9]
```diff
@@ -0,0 +1,111 @@
+import 'package:ambush_app/src/core/routes/app_route.gr.dart';
+import 'package:ambush_app/src/core/settings/const.dart';
+import 'package:auto_route/auto_route.dart';
+import 'package:ambush_app/src/designsystem/buttons.dart';
+import 'package:ambush_app/src/presentation/add_invoice/add_invoice_navigation_flow.dart';
+import 'package:flutter/material.dart';
+
+import '../../../designsystem/constrained_scaffold.dart';
```
Comment (theolm) [inline lib/src/presentation/settings/success/onboarding_success_screen.dart]: Don't use the relative path in the import unless is an autogenerated file.

**Review by theolm — CHANGES_REQUESTED** [review]
The component is missing some attributes and the pages still have different behaviour.

The new screen has a constrained size while the others have an infinity width.

File: lib/src/designsystem/constrained_scaffold.dart (hunk 6 above)
Comment (alexrodrigues) [reply lib/src/designsystem/constrained_scaffold.dart]: Got it! I've circled back and included all the default Scaffold into the ConstrainedScaffold component.
This way the Engineer can customize the Constrained Scaffold the way he wants

[commit](https://github.com/GetAmbush/ambush-app/pull/84/commits/862c5b98b99ae91fb9b2eba19fa59fb20cbd558e)

File: lib/src/presentation/settings/base_settings_page.dart (hunk 7 above)
Comment (alexrodrigues) [reply lib/src/presentation/settings/base_settings_page.dart]: My mistake, I’ve gone back and changed it to use defaultPageMaxWidth

[Commit](https://github.com/GetAmbush/ambush-app/pull/84/commits/862c5b98b99ae91fb9b2eba19fa59fb20cbd558e)

File: lib/src/designsystem/constrained_scaffold.dart (hunk 8 above)
Comment (alexrodrigues) [reply lib/src/designsystem/constrained_scaffold.dart]: Agreed! I've just implemented your suggestions (the maxWidth is no longer required and its default value is double.infinity):

```
  const ConstrainedScaffold({
    super.key,
    this.appBar,
    this.maxWidth = double.infinity,
```

[Commit](https://github.com/GetAmbush/ambush-app/pull/84/commits/862c5b98b99ae91fb9b2eba19fa59fb20cbd558e)

File: lib/src/presentation/settings/success/onboarding_success_screen.dart (hunk 9 above)
Comment (alexrodrigues) [reply lib/src/presentation/settings/success/onboarding_success_screen.dart]: Good to know! I've just fixed that too

[Commit](https://github.com/GetAmbush/ambush-app/pull/84/commits/862c5b98b99ae91fb9b2eba19fa59fb20cbd558e)

**Comment by alexrodrigues** [comment]
Thanks for your feedback @theolm!

**Review by theolm — APPROVED** [review]
Nice job!

**Pull request merged**